File: backend/app/models/policy.py
from pydantic import BaseModel, Field


class PolicyDocument(BaseModel):
    id: str
    title: str
    authority: str
    document_number: str | None = None
    category: str
    published_date: str
    effective_date: str | None = None
    status: str
    source_url: str
    updated_at: str
    keywords: list[str]
    summary: str
    content: str


class PolicySearchRequest(BaseModel):
    query: str = Field(min_length=1)
    category: str | None = None
    limit: int = Field(default=5, ge=1, le=20)


class PolicySearchResult(BaseModel):
    document: PolicyDocument
    relevance_score: float
    snippets: list[str]


class PolicySearchResponse(BaseModel):
    query: str
    total: int
    results: list[PolicySearchResult]
    latest_policy_check_required: bool = True

File: frontend/src/types/policy.ts
export interface PolicyDocument {
  id: string;
  title: string;
  authority: string;
  document_number: string | null;
  category: string;
  published_date: string;
  effective_date: string | null;
  status: string;
  source_url: string;
  updated_at: string;
  keywords: string[];
  summary: string;
  content: string;
}

export interface PolicySearchResult {
  document: PolicyDocument;
  relevance_score: number;
  snippets: string[];
}

export interface PolicySearchResponse {
  query: string;
  total: number;
  results: PolicySearchResult[];
  latest_policy_check_required: boolean;
}
